### PR TITLE
Update Nek5000 spack package to release 17.0

### PR DESCRIPTION
**Pull request by thilinarmtb**
Just updated the Nek5000 `spack` package to use the stable release 17 (instead of the beta). 
This is ready to be merged as it is if the reviewers are fine with this.

PS: I don't know why this PR shows all the previous commit history even though actual
diff is just the last two commits. May be because I reused my older local git branch?

### CODE REVIEW

File: var/spack/repos/builtin/packages/nek5000/package.py [hunk 1]
```diff
@@ -46,9 +46,9 @@ class Nek5000(Package):
     tags = ['cfd', 'flow', 'hpc', 'solver', 'navier-stokes',
             'spectral-elements', 'fluid']
 
-    version('17.0.0-beta2', git='https://github.com/Nek5000/Nek5000.git',
-        commit='b95f46c59f017fff2fc19b66aa65a881085a7572')
-    version('develop',      git='https://github.com/Nek5000/Nek5000.git',
+    version('17.0',    git='https://github.com/Nek5000/Nek5000.git',
+        commit='469daf94d3f9aa3ba9d258d8eee9ebde6893a702', preferred=True)
```
Comment (davydden) [inline var/spack/repos/builtin/packages/nek5000/package.py]: No need for preferred, it’ll be used by default

Comment (thilinarmtb) [reply var/spack/repos/builtin/packages/nek5000/package.py]: Fixed, Thanks !!

**Comment by thilinarmtb** [comment]
Code coverage tests fail. Do I need to fix that?

**Comment by thilinarmtb** [comment]
I am okay for merging this if there are no additional comments.
It will be great if this can be merged soon as this is part of CEED
software suite.

**Comment by thilinarmtb** [comment]
@acfisher: Once this PR is merged,  we need to update CEED 1.0 release to use version 17.0
of Nek5000, not the beta version.

**Comment by thilinarmtb** [comment]
@tgamblin @adamjstewart It will be great to get this merged as this is going to be part of CEED 1.0 release.

**Comment by adamjstewart** [comment]
> Code coverage tests fail. Do I need to fix that?

See the note at the bottom of http://spack.readthedocs.io/en/latest/contribution_guide.html#coverage

File: var/spack/repos/builtin/packages/nek5000/package.py [hunk 2]
```diff
@@ -88,17 +88,14 @@ def install(self, spec, prefix):
...
         # Build the tools, maketools copy them to Nek5000/bin by default.
         # We will then install Nek5000/bin under prefix after that.
         with working_dir(toolsDir):
             # Update the maketools script to use correct compilers
-            if self.version == Version('17.0.0-beta2'):  # Old release
-                filter_file(r'^F77\s*=.*', 'F77=\"' + F77 + '\"', 'maketools')
-            else:
-                filter_file(r'^FC\s*=.*', 'FC=\"' + F77 + '\"', 'maketools')
+            filter_file(r'^FC\s*=.*', 'FC=\"' + FC + '\"', 'maketools')
             filter_file(r'^CC\s*=.*', 'CC=\"' + CC  + '\"',   'maketools')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/nek5000/package.py]: You don't need to escape these `'"'`. Same below. Personally, I would use `'CC="{0}"'.format(CC)'`, but this is fine too.

Comment (thilinarmtb) [reply var/spack/repos/builtin/packages/nek5000/package.py]: Thanks Adam !! I will fix this.

**Comment by thilinarmtb** [comment]
@adamjstewart: Thanks for the reviews !!

**Comment by adamjstewart** [comment]
Also, 107 commits and 136 changed files kinda scares me. You should create a new branch that only includes the changes you intend to make.

**Comment by thilinarmtb** [comment]
Yep, I was pushing to my old branch. That was the problem. I will close this and send a new PR.

**Pull request closed without merging**